mibios/glamr/views.py

```diff
@@ -493,14 +501,86 @@ def get_details(self):
             pk = 0
         else:
             pk = self.object.pk
-        # prepend samples row
-        details = [(
-            'Samples',
-            reverse('dataset_sample_list', args=[pk]),
-            f'List of {self.object.samples().count()} samples'
-        )] + details
-        return details, rel_lists
 
+        # this is hacky but want details in particular order
+        ordered_details = []
+        # reference, samples, bioproject, gold id, material type, water bodies,
+        # primers, sequencing target, sequencing platform, size fraction, notes
+        for detail in details:
+            # reference
+            if detail[0] == 'reference':
+                if detail[2]:
+                    ordered_details.append((
+                        'Reference',
+                        detail[1],
+                        detail[2]
+                    ))
+                ordered_details.append((
+                    'Samples',
+                    reverse('dataset_sample_list', args=[pk]),
+                    f'{self.object.samples().count()} available samples'
+                ))
+            if detail[0] == 'bioproject' and detail[2]:
+                ordered_details.append((
+                    'Bio Project',
+                    detail[1],
+                    detail[2]
+                ))
+            if detail[0] == 'gold id' and (detail[1] or detail[2]):
+                ordered_details.append((
+                    'Gold ID',
+                    detail[1],
+                    detail[2]
+                ))
+            if detail[0] == 'material type' and detail[2]:
+                value_text = detail[2]
+                if value_text.lower() == 'edna':
+                    value_text = "eDNA"
+                else:
```

Comment:
for those things (there are others) I'll add a template tag, "capfirstkeep" that will work like capfirst except if the value contains an upper case letter already, then it keeps the exoisting value.